ambari-server/src/main/python/ambari_server/setupSecurity.py

```diff
@@ -278,7 +281,7 @@ def sync_ldap(options):
 
   properties = get_ambari_properties()
 
-  if get_value_from_properties(properties,CLIENT_SECURITY_KEY,"") == 'pam':
+  if get_value_from_properties(properties,CLIENT_SECURITY,"") == 'pam':
```

Comment:
This block should probably be more generic.  Rather than checking for PAM, it should ensure the type is LDAP and fail if it is not.  In the future, PAM may not be the only alternative.